src/language-html/print/block.js

```diff
@@ -0,0 +1,123 @@
+import { group, hardline, indent } from "../../document/builders.js";
+import { printChildren } from "./children.js";
+
+let uid = 0;
+
+const settings = {
+  if: {
+    withExpression: true,
+    isFollowingBlock: false,
+    followingBlocks: ["else if", "else"],
+  },
+  "else if": {
+    withExpression: true,
+    isFollowingBlock: true,
+    followingBlocks: ["else if", "else"],
+  },
+  else: {
+    withExpression: false,
+    isFollowingBlock: true,
+    followingBlocks: [],
+  },
+
+  switch: {
+    withExpression: true,
+    isFollowingBlock: false,
+    followingBlocks: [],
+  },
+  case: {
+    withExpression: true,
+    isFollowingBlock: false,
+    followingBlocks: [],
+  },
+  default: {
+    withExpression: false,
+    isFollowingBlock: false,
+    followingBlocks: [],
+  },
+
+  for: {
+    withExpression: true,
+    isFollowingBlock: false,
+    followingBlocks: ["empty"],
+  },
+  empty: {
+    withExpression: false,
+    isFollowingBlock: true,
+    followingBlocks: [],
+  },
+
+  defer: {
+    withExpression: true,
+    isFollowingBlock: false,
+    followingBlocks: ["placeholder", "error", "loading"],
+  },
+  placeholder: {
+    withExpression: false,
```

Comment:
Can this `withExpression` get from the AST?

`node.parameters`

Reply:
Expression is required under the `if` and `for`
I changed the variable to "shouldExpression" to be more clear